### PR TITLE
Make logfire as an optional dependency

### PR DESCRIPTION
**Pull request by hlloreda**
# Description

- `Logfire` is now an optional dependency

closes https://github.com/inmanta/inmanta-core/issues/8154

# Self Check:

Strike through any lines that are not applicable (`~~line~~`) then check the box

- [ ] Attached issue to pull request
- [ ] Changelog entry
- [ ] Type annotations are present
- [ ] Code is clear and sufficiently documented
- [ ] No (preventable) type errors (check using make mypy or make mypy-diff)
- [ ] Sufficient test cases (reproduces the bug/tests the requested feature)
- [ ] Correct, in line with design
- [ ] End user documentation is included or an issue is created for end-user documentation (add ref to issue here: )
- [ ] If this PR fixes a race condition in the test suite, also push the fix to the relevant stable branche(s) (see [test-fixes](https://internal.inmanta.com/development/core/tasks/build-master.html#test-fixes) for more info)

### CODE REVIEW

File: requirements.dev.txt [hunk 1]
```diff
@@ -1,5 +1,6 @@
 inmanta-dev-dependencies[pytest,async,core]==2.137.0
 bumpversion==0.6.0
+#-e .[tracing]
```
Comment (hlloreda) [inline requirements.dev.txt:3]: This needs to be uncommented (only used for testing purpose)

File: src/inmanta/tracing.py [hunk 2]
```diff
@@ -19,21 +19,79 @@
...
+    import logfire._internal.config
+    import logfire.integrations
+    import logfire.integrations.pydantic
+    from logfire import LevelName, LogfireSpan
+    from logfire._internal.main import NoopSpan
+    from logfire.propagate import ContextCarrier
+
+    # Make sure we don't get warnings when it is off
+    logfire._internal.config.GLOBAL_CONFIG.ignore_no_config = True
+else:
```
Comment (hlloreda) [inline src/inmanta/tracing.py:41]: I'm a bit on the fence if this or the try except should be better. Any Opinion?

I'm a bit worried that with a `try except` approach, we try to enable logging even though the dependencies are not installed. We could still assert that the env var is not set if the library is not present

File: tests/conftest.py [hunk 3]
```diff
@@ -1983,13 +1982,3 @@ def disable_version_and_agent_cleanup_job():
...
     orchestrationservice.PERFORM_CLEANUP = old_perform_cleanup
-
-
-@pytest.fixture(scope="session", autouse=not PYTEST_PLUGIN_MODE)
-def configure_logfire():
-    """Configure logfire to ensure all the instrumentation works correctly and does not provide warnings. This does not
-    setup tracing for tests"""
-    logfire.configure(
-        send_to_logfire=False,
-        console=False,
```
Comment (hlloreda) [inline tests/conftest.py:1994]: I don't think we need to keep this? As we are already calling `configure_logfire` in multiple places (components)

File: changelogs/unreleased/8154-optional-logfire-dependency.yml
Comment (hlloreda) [inline changelogs/unreleased/8154-optional-logfire-dependency.yml:1]: test in another venv (or discuss it further)

File: src/inmanta/tracing.py (hunk 2 above)
Comment (hlloreda) [reply src/inmanta/tracing.py:41]: Change to try and except, we should not crash if this token was used for something else

File: changelogs/unreleased/8154-optional-logfire-dependency.yml
Comment (hlloreda) [reply changelogs/unreleased/8154-optional-logfire-dependency.yml:1]: Let it because mypy and use "TYPING"

File: src/inmanta/tracing.py [hunk 4]
```diff
@@ -82,16 +137,16 @@ def span(
     *,
     _tags: Sequence[str] | None = None,
     _span_name: str | None = None,
-    _level: LevelName | None = None,
+    _level: LevelName | None | "logfire.LevelName" = None,
```
Comment (hlloreda) [inline src/inmanta/tracing.py:140]: This does not work, any suggestion?

File: setup.py [hunk 5]
```diff
@@ -15,7 +15,6 @@
     "docstring-parser>=0.10,<0.17",
     "email-validator>=1,<3",
     "jinja2~=3.0",
-    "logfire>=0.46,<2.0",
     "more-itertools>=8,<11",
     "opentelemetry-instrumentation-asyncpg~=0.46b0",
```
Comment (wouterdb) [inline setup.py:19]: this one has to be in the extra as well

File: src/inmanta/tracing.py (hunk 4 above)
Comment (wouterdb) [reply src/inmanta/tracing.py:140]: what doesn't work/ how does it nor work?

File: tox.ini [hunk 6]
```diff
@@ -8,8 +8,8 @@ requires = pip
 # We need this to be allowed to use the timeout command
 allowlist_externals=timeout
 deps=
+    .
     -rrequirements.dev.txt
-    -rrequirements.txt
```
Comment (wouterdb) [inline tox.ini:12]: ?

**Review by wouterdb — CHANGES_REQUESTED** [review]

File: tox.ini (hunk 6 above)
Comment (hlloreda) [reply tox.ini:12]: I thought it was enough to install inmanta-core using `.` (setup.py)

**Pull request closed without merging**